multiapps-controller-web/src/main/java/org/cloudfoundry/multiapps/controller/web/api/impl/FilesApiServiceImpl.java

```diff
@@ -181,21 +194,38 @@ private FileEntry uploadFileFromUrl(String spaceGuid, String namespace, String f
         }
     }
 
-    protected HttpClient buildHttpClient() {
+    protected HttpClient buildHttpClient(String decodedUrl) {
         return HttpClient.newBuilder()
-                         //ssl and authentication configuration can be done here
                          .version(HttpClient.Version.HTTP_1_1)
                          .connectTimeout(Duration.ofMinutes(10))
                          .followRedirects(Redirect.NORMAL)
+                         .authenticator(new Authenticator() {
+                             @Override
+                             protected PasswordAuthentication getPasswordAuthentication() {
+                                 if (decodedUrl.contains("@")) {
+                                     var uri = URI.create(decodedUrl);
+                                     var userInfo = uri.getUserInfo();
+                                     var userPassParts = userInfo.split(":");
```

Comment:
Do not split it! The password can contain ":", the username cannot. Use substring.